lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -58,9 +59,19 @@ llvm::raw_ostream &printHash(llvm::raw_ostream &stream, std::string data) {
   return printHex(stream, bytes);
 }
 
-struct StructuralHasher {
-  explicit StructuralHasher(MLIRContext *context) {
+// This data structure is used as hash for modules. A first element is a
+// structural hash of the module, and the second element is an array which
+// tracks module names encountered in the walk. Since module names could
+// be replaced during dedup, it's necessary to keep names up-to-date before
+// actually combining them into structural hashes.
+using Hash = std::pair<std::array<uint8_t, 32>, mlir::Attribute>;
```

Comment:
I'm not sure it makes sense to call this a Hash anymore. Would it be worth creating a struct for this?

Reply:
That makes sense, created `ModuleInfo` struct which is a pair of std::array and SmallVector.